src/main/java/com/mabl/integration/jenkins/test/output/Properties.java

```diff
@@ -1,28 +1,45 @@
 package com.mabl.integration.jenkins.test.output;
 
+import com.fasterxml.jackson.databind.annotation.JsonAppend;
 import com.google.common.collect.ImmutableCollection;
 
 import javax.xml.bind.annotation.XmlAccessType;
 import javax.xml.bind.annotation.XmlAccessorType;
 import javax.xml.bind.annotation.XmlElement;
 import javax.xml.bind.annotation.XmlRootElement;
+import java.util.ArrayList;
+import java.util.Collection;
 
 @XmlRootElement(name = "properties")
 @XmlAccessorType(XmlAccessType.FIELD)
 public class Properties {
 
     @XmlElement(name = "property")
-    private ImmutableCollection<Property> properties;
+    private Collection<Property> properties;
 
-    public Properties(ImmutableCollection<Property> properties) {
+    public Properties(Collection<Property> properties) {
         this.properties = properties;
     }
 
     public Properties() {
 
     }
 
-    public ImmutableCollection<Property> getProperties() {
+    public Collection<Property> getProperties() {
```

Comment:
Do we want people to be able to mutate these?

Reply:
Is the use case to remove properties? I ask, since it looks like helpers were added below to allow adding properties, which seems redundant with removing the defensive copy.

Reply:
Frankly, this is only used in unit tests. I prefer not using external classes when not necessary. I just much prefer returning a Collections.unmodifiableCollection(properties) instead. This class is only used when we are serializing JUnit XML report.